src/views/virtualmachines/details/components/snapshots/SnapshotRow.tsx

```diff
@@ -0,0 +1,48 @@
+import * as React from 'react';
+
+import {
+  V1alpha1VirtualMachineRestore,
+  V1alpha1VirtualMachineSnapshot,
+} from '@kubevirt-ui/kubevirt-api/kubevirt';
+import { RowProps, TableData } from '@openshift-console/dynamic-plugin-sdk';
+
+import IndicationLabelList from './components/IndicationLabel/IndicationLabelList';
+import { useSnapshotStatus } from './hooks/useSnapshotStatus';
+import Timestamp from '../../../list/components/Timestamp/Timestamp';
+
+const SnapshotRow: React.FC<
+  RowProps<V1alpha1VirtualMachineSnapshot, { restores: V1alpha1VirtualMachineRestore[] }>
+> = ({ obj: snapshot, activeColumnIDs, rowData: { restores } }) => {
+  const relevantRestore = restores?.find(
```

Comment:
this could have bad performance in scale.
please create a map of restores, where the key is `snapshot?.metadata?.name`

Reply:
changed, thanks :)